### PR TITLE
feat: theme system 2.0

### PR DESCRIPTION
**Pull request by kyhyco**
**What changed? Why?**
- Remove all Text components and replaced with `text` theme object that contains all styles

**Notes to reviewers**

**How has it been tested?**
locally

### CODE REVIEW

File: src/internal/text/TextAs.tsx [hunk 1]
```diff
@@ -1,25 +1,23 @@
...
+  color = 'text-foreground',
+  defaultClassName,
   ...props
 }: TextAsReact) {
-  const textColor = textColorMap[color];
-
   return createElement(
     as,
-    { className: cn(className, textColor), ...props },
+    { className: cn(defaultClassName, color, className), ...props },
```
Comment (kyhyco) [inline src/internal/text/TextAs.tsx:20]: This will determine the style merging order.

File: src/internal/text/types.ts [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
-  color?: keyof typeof textColorMap;
+  color?:
+    | 'text-inverse'
+    | 'text-foreground'
+    | 'text-foreground-muted'
+    | 'text-error'
+    | 'text-primary'
+    | 'text-success'
+    | 'text-warning'
+    | 'text-disabled';
```
Comment (kyhyco) [inline src/internal/text/types.ts:14]: Instead of mapping, directly using allowed text color class names.

File: src/styles/theme.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+export const text = {
+  body: 'font-sans text-foreground text-base leading-normal',
+  caption: 'font-sans text-foreground text-bold text-xs leading-4',
+  headline: 'font-bold text-foreground font-sans text-base leading-normal',
+  label1: 'font-bold text-foreground font-sans text-sm leading-5',
+  label2: 'font-sans text-foreground text-sm leading-5',
+  legal: 'font-sans text-foreground text-xs leading-4',
+  title3: 'font-bold font-display text-xl leading-7',
+} as const;
```
Comment (kyhyco) [inline src/styles/theme.ts:16]: `as const` enables devs to see what the styles are by looking at the definition

**Review by Zizzamia — APPROVED** [review]

**Review by ilikesymmetry — COMMENTED** [review]
Love love love this!

**Pull request merged**